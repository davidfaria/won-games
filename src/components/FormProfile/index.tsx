import Button from 'components/Button'
import Heading from 'components/Heading'
import TextField from 'components/TextField'
import * as S from './styles'

const FormProfile = () => (
  <>
    <Heading lineBottom color="black" lineColor="primary">
      My Profile
    </Heading>

    <S.Form>
      <TextField name="name" label="Name" initialValue="David Faria" />
      <TextField
        type="email"
        name="email"
        label="E-mail"
        initialValue="davidfaria89@gmail.com"
        disabled
      />
      <TextField
        type="password"
        name="password"
        label="Password"
        placeholder="Type your password"
      />

      <TextField
        type="password"
        name="password"
        label="New password"
        placeholder="New password"
      />

      <Button>Save</Button>
    </S.Form>
  </>
)

export default FormProfile
