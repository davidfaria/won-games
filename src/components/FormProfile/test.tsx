import { screen } from '@testing-library/react'
import { renderWithTheme } from 'utils/tests/helpers'

import FormProfile from '.'

describe('<FormProfile />', () => {
  it('should render the form profile', () => {
    const { container } = renderWithTheme(<FormProfile />)

    expect(
      screen.getByRole('heading', { name: /My profile/i })
    ).toBeInTheDocument()

    expect(screen.getByRole('textbox', { name: /name/i })).toBeInTheDocument()

    expect(screen.getByRole('textbox', { name: /e-mail/i })).toBeInTheDocument()

    expect(
      screen.getByPlaceholderText('Type your password')
    ).toBeInTheDocument()

    expect(screen.getByPlaceholderText('New password')).toBeInTheDocument()

    expect(screen.getByRole('button', { name: /save/i })).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })
})
