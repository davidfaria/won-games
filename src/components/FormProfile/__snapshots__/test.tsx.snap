// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormProfile /> should render the form profile 1`] = `
.c0 {
  color: #030517;
  position: relative;
  margin-bottom: 3.2rem;
  font-size: 2.0rem;
}

.c0::after {
  position: absolute;
  left: 0;
  bottom: -0.5rem;
  content: '';
  width: 5rem;
  border-bottom: 0.5rem solid #F231A5;
}

@media (min-width:768px) {
  .c0 {
    font-size: 2.8rem;
  }
}

@media (min-width:768px) {

}

<h2
  class="c0"
  color="black"
>
  My Profile
</h2>
`;
